### PR TITLE
[FFP-460] Files, Session Auth and More

### PR DESCRIPTION
**Pull request by miguelnmiranda**
Fixes FFP-460 FFP-473

- Files API
- Sessions API
- Misc.

## Checklist

- [ ] Checked the work against _all_ of the ticket requirements
- [ ] CHANGELOG updated
- [ ] Formatter has been run against all changed files
- [ ] All changed files have been added correctly
- [ ] Appropriate unit tests and integration tests have been added
- [ ] All tests are passing
- [ ] Ad hoc testing has been done
- [ ] Affected docs have been updated
- [ ] Tickets created for any TODOs
- [ ] All licenses have been checked and confirmed for commercial use

## Back End Specific

- [ ] Database migrations have been tested
- [ ] Necessary data patches have been prepared
- [ ] Infrastructure changes have been completed

### CODE REVIEW

**Review by antoniofpnmartins — APPROVED** [review]

File: api/alembic/env.py [hunk 1]
```diff
@@ -13,6 +13,9 @@
 import recursiveai.findflow.assistant.api.models.user_groups
 import recursiveai.findflow.assistant.api.models.blocked_keywords
 import recursiveai.findflow.assistant.api.models.conversations
+import recursiveai.findflow.assistant.api.models.files
+# import recursiveai.findflow.assistant.api.models.metrics
+# import recursiveai.findflow.assistant.api.models.audits
```
[the remark this answers is not in the document]
Comment (miguelnmiranda) [reply api/alembic/env.py:18]: Yes, only when implementing the service/api will have it created

File: api/src/recursiveai/findflow/assistant/common/dependencies.py [hunk 2]
```diff
@@ -36,12 +37,14 @@ def singleton(func: Callable[P, T]) -> Callable[P, Awaitable[T]]:
...
     async def singleton_decorator(*args, **kwargs) -> T:
-        async with lock:
-            if func not in _store_registry:
-                _logger.debug("Creating Singleton %s", func.__name__)
-                _store_registry[func] = func(*args, **kwargs)
-            else:
-                _logger.debug("Reusing Singleton %s", func.__name__)
+        if func not in _store_registry:
+            async with lock:
+                if func not in _store_registry:
```
[the remark this answers is not in the document]
Comment (miguelnmiranda) [reply api/src/recursiveai/findflow/assistant/common/dependencies.py:42]: applying https://en.wikipedia.org/wiki/Double-checked_locking pattern given after the init doing the lock should be unnecessary..

File: api/src/recursiveai/findflow/assistant/common/dependencies.py [hunk 3]
```diff
@@ -52,12 +55,14 @@ def async_singleton(func: Callable[P, Awaitable[T]]) -> Callable[P, Awaitable[T]
...
     async def singleton_decorator(*args, **kwargs) -> T:
-        async with lock:
-            if func not in _store_registry:
-                _logger.debug("Creating AsyncSingleton %s", func.__name__)
-                _store_registry[func] = await func(*args, **kwargs)
-            else:
-                _logger.debug("Reusing Async Singleton %s", func.__name__)
+        if func not in _store_registry:
+            async with lock:
+                if func not in _store_registry:
```
[the remark this answers is not in the document]
Comment (miguelnmiranda) [reply api/src/recursiveai/findflow/assistant/common/dependencies.py:60]: as before

File: api/src/recursiveai/findflow/assistant/api/app.py [hunk 4]
```diff
@@ -38,6 +42,13 @@ def create_app(lifespan_context=None) -> FastAPI:
         lifespan=lifespan_context,
     )
 
+    # TODO: Review Implication of using it everywhere
+    config = AppConfig()  # await app_config()
```
[the remark this answers is not in the document]
Comment (miguelnmiranda) [reply api/src/recursiveai/findflow/assistant/api/app.py:46]: Was looking into that yesterday.. the issue was with using a thread lock vs async lock in the singleton..

**Pull request merged**